### ISSUE TITLE
🎨 ui-selected and ui-option are inconsistent

### ISSUE THREAD

**Issue by SRWieZ** [livewire/flux#357]
TLDR: The selected option doesn't show the same when selected because there is a `gap-2` on ui-selected but not on `flux:option`.

If you have time, we had a nice (and relatively short) debate on how to resolve the problem in the following comments.

Screenshot of the problem:

![Image](https://github.com/user-attachments/assets/6d655424-59c1-4612-8c18-b665848d9eb6)

Code to reproduce it:
```blade
<flux:select variant="listbox" label="Deployment tool" >
  <flux:option>
    <img src="https://forge.laravel.com/favicon-32x32.png" class="size-5 flex-shrink-0 rounded-full">
    <span>Forge</span>
  </flux:option>
  <flux:option>
    <img src="https://envoyer.io/img/favicons/favicon-32x32.png" class="size-5 flex-shrink-0 rounded-full">
    <span>Evoyer.io</span>
  </flux:option>
</flux:select>
```

Possible solutions:  
-  Add a `gap-2` in `flux:option` as an opinionated design choice. User code stays the same, and the gap is forced.  
-  Wrap the selected option with a div, as the `gap-2` is probably there for another reason. User will be responsible for putting the gap where they want it.

**Comment by ju5t** [comment]
I don't think there should be a `gap-2` there.  You're in control over what gets passed to a Blade slot. You could just do:

```blade
<flux:select variant="listbox" label="Deployment tool" >
    <flux:option>
        <div class="gap-2">
            <img src="https://forge.laravel.com/favicon-32x32.png" class="size-5 flex-shrink-0 rounded-full">
            <span>Forge</span>
        </div>
    </flux:option>
    ...
</flux:select>
```

This is not up to Flux.

**Comment by SRWieZ** [comment]
I can also put `class="gap-2"` manually on `flux:option`.

But since we have it on ui-selected, adding it to ui-option will be great for [Brevity](https://fluxui.dev/docs/principles#brevity) and [Consistency](https://fluxui.dev/docs/principles#consistency).

I don't see any drawbacks to adding `gap-2` on Flux directly. It's worth considering.

**Comment by SRWieZ** [comment]
Still, it's inconsistent.

Let's say I didn't want this gap with my image. Why is there a gap in ui-selected and not in ui-option? Maybe the solution is to remove the gap-2 from ui-selected or to wrap the selected element in a div so there is no gap in ui-option and ui-selected.

Either way, I shouldn't have to add a div to flux:option so it looks consistent between the ui-option and ui-selected elements.

**Comment by ju5t** [comment]
> I shouldn't have to add a div to flux:option so it looks consistent between the ui-option and ui-selected elements.

I agree, `ui-selected` shouldn't have `gap-2` either.

**Comment by SRWieZ** [comment]
I don't mind the `gap-2` because I see it as an opinionated design choice that I'm fine with.

But you're also right; maybe it should be my responsibility to address the gap with a surrounding div.

Anyway, thanks for the argument/discussion. I will update my first post to better explain the problem.

**Comment by calebporzio** [comment]
Yeah, unfortunately, I've been down this road myself. The best thing to do is wrap your option contents in a <div> so that you have full control over things like spacing. Like @ju5t  said in this comment: https://github.com/livewire/flux/issues/357#issuecomment-2400248510
